### PR TITLE
feat: upgrade node mailjet version

### PR DESCRIPTION
**Pull request by scopsy**
### What change does this PR introduce?

Upgrade to the latest node mailjet version

### Why was this change needed?

Old version had a vulnerable `vm2` package, we needed to upgrade.

### CODE REVIEW

File: providers/mailjet/src/lib/mailjet.provider.ts [hunk 1]
```diff
@@ -8,16 +8,15 @@ import {
   IEmailEventBody,
   EmailEventStatusEnum,
 } from '@novu/stateless';
-import Client, { Email } from 'node-mailjet';
-import { MailjetResponse } from './mailjet-response.interface';
```
Comment (djabarovgeorge) [inline providers/mailjet/src/lib/mailjet.provider.ts:12]: can we delete this interface or someone use it?

File: providers/mailjet/src/lib/mailjet.provider.ts [hunk 2]
```diff
@@ -26,24 +25,28 @@ export class MailjetEmailProvider implements IEmailProvider {
...
-    const response = (await send.request(
-      requestObject
-    )) as unknown as MailjetResponse;
+    const { body, response: clientResponse } = response;
 
     return {
-      id: response.response.header['x-mj-request-guid'],
-      date: response.response.header.date,
+      id: clientResponse.headers['x-mj-request-guid'],
+      date: new Date().toISOString(),
```
Comment (djabarovgeorge) [inline providers/mailjet/src/lib/mailjet.provider.ts:49]: i guess we are missing the provider date

**Review by djabarovgeorge — APPROVED** [review]
✨

i wonder if there is a good way to test e2e third-party packages/services

File: providers/mailjet/src/lib/mailjet.provider.ts (hunk 2 above)
Comment (scopsy) [reply providers/mailjet/src/lib/mailjet.provider.ts:49]: Yep :(

**Comment by scopsy** [comment]
@djabarovgeorge currently made the manual tests for this with mailjet api key.

**Pull request merged**